WRT Regressions #3:

The new snowball-specific test `romancal/ramp_fitting/tests/test_ramp_fit_likelihood.py::test_flag_large_events_withsnowball` succeeds when run locally on an M4 architecture. However, the GA version, rather significantly, fails:
```
        # Check that the uncertainties are the same for all pixels with jumps
        # and save this average uncertainty
        meanerr_jumppixels_new = np.mean(m_image.err[m_image.dq == pixel.JUMP_DET])
>       assert np.std(m_image.err[m_image.dq == pixel.JUMP_DET]) < 1e-6
E       assert np.float16(0.000546) < 1e-06
```
Unsure of what to make of that, except that this maybe what has been seen in the past where minor precision changes upstream cause dramatic changes in the ramp fitting.

--- a/romancal/pipeline/tests/test_exposure_pipeline.py
+++ b/romancal/pipeline/tests/test_exposure_pipeline.py
@@ -70,13 +70,13 @@ def test_elp_save_results(function_jail, save_results, monkeypatch):
 
     model = rdm.ScienceRawModel.create_fake_data()
     model.meta.filename = "test_uncal.asdf"
-    model.meta.exposure.read_pattern = [[1], [2]]  # truncated for 2 groups below
+    model.meta.exposure.read_pattern = [[1], [2], [3], [4]]  # truncated for 4 groups below
     model.meta.exposure.ma_table_number = 5
     model.meta.exposure.start_time = Time(
         "2024-01-03T00:00:00.0", format="isot", scale="utc"
     )
-    model.data = np.zeros((2, 4096, 4096), dtype=model.data.dtype)
-    model.amp33 = np.zeros((2, 4096, 128), dtype=model.amp33.dtype)
+    model.data = np.zeros((4, 4096, 4096), dtype=model.data.dtype)
+    model.amp33 = np.zeros((4, 4096, 128), dtype=model.amp33.dtype)
 
     pipeline = ExposurePipeline()
     pipeline.output_dir = str(output_path)

--- a/romancal/ramp_fitting/ramp_fit_step.py
+++ b/romancal/ramp_fitting/ramp_fit_step.py
@@ -37,7 +37,7 @@ class RampFitStep(RomanStep):
     class_alias = "ramp_fit"
 
     spec = """
-        algorithm = option('ols_cas22', 'likely', default='ols_cas22')  # Algorithm to use to fit.
+        algorithm = option('ols_cas22', 'likely', default='likely')  # Algorithm to use to fit.
         suffix = string(default='rampfit')  # Default suffix of results
         use_ramp_jump_detection = boolean(default=True) # Use jump detection during ramp fitting
         threshold_intercept = float(default=None) # Override the intercept parameter for the threshold function in the jump detection algorithm.

--- a/romancal/ramp_fitting/tests/common.py
+++ b/romancal/ramp_fitting/tests/common.py
@@ -35,6 +35,18 @@ SIMPLE_RESULTANTS = np.array(
 # #########
 # Utilities
 # #########
+
+def create_linear_ramp(n_resultants=4, nrows=2, ncols=2, rate=2, pedistal=1):
+    """Create linear resultants"""
+    r1 = np.array(list(range(n_resultants)), dtype=np.float32) * rate + pedistal
+    r2 = r1[:, np.newaxis]
+    r2 = np.repeat(r2, repeats=nrows, axis=1)
+    ramps = r2[:, np.newaxis]
+    ramps = np.repeat(ramps, repeats=ncols, axis=1)
+
+    return ramps
+
+
 def make_data(resultants, ingain, rnoise, randomize):
     """Create test input data
 

--- a/romancal/ramp_fitting/tests/test_ramp_fit_likelihood.py
+++ b/romancal/ramp_fitting/tests/test_ramp_fit_likelihood.py
@@ -4,8 +4,9 @@ import numpy as np
 import pytest
 
 from romancal.ramp_fitting import RampFitStep
+from roman_datamodels.dqflags import pixel
 
-from .common import SIMPLE_RESULTANTS, make_data
+from .common import SIMPLE_RESULTANTS, create_linear_ramp, make_data
 
 SIMPLE_EXPECTED_DEFAULT = {
     "data": np.array(
@@ -75,6 +76,65 @@ def test_bad_readpattern():
             override_gain=gain_model,
             override_readnoise=readnoise_model,
         )
+
+
+def test_flag_large_events_withsnowball():
+    """Test that large events are flagged"""
+    resultants = create_linear_ramp(n_resultants=20, nrows=100, ncols=100)
+    model, m_gain, m_rnoise, m_dark = make_data(resultants, 6, 0.01, False)
+
+    # square of saturation surrounded by jump -> snowball
+    # 112 pixels (121 minus 9) initially have a jump.
+    model.data[10:, 46:57, 46:57] += 300
+    model.data[10:, 50:53, 50:53] = 1e5
+    model.groupdq[10:, 50:53, 50:53] = pixel.SATURATED
+
+    m_image = RampFitStep.call(
+        model,
+        algorithm='likely',
+        override_gain=m_gain,
+        override_readnoise=m_rnoise,
+        expand_large_events = True,
+        min_sat_area = 1,
+        min_jump_area = 6,
+        expand_factor = 1.9,
+        edge_size = 0,
+        sat_required_snowball = True,
+        min_sat_radius_extend = 0.5,
+        sat_expand = 2,
+    )
+    assert np.std(m_image.data) < 1e-5
+    n_jump_expanded = np.sum(m_image.dq == pixel.JUMP_DET)
+
+    # Check that the uncertainties are the same for all pixels with jumps
+    # and save this average uncertainty
+    meanerr_jumppixels_new = np.mean(m_image.err[m_image.dq == pixel.JUMP_DET])
+    assert np.std(m_image.err[m_image.dq == pixel.JUMP_DET]) < 1e-6
+
+    # Now without snowball flagging
+    model, m_gain, m_rnoise, m_dark = make_data(resultants, 6, 0.01, False)
+    model.data[10:, 46:57, 46:57] += 300
+    model.data[10:, 50:53, 50:53] = 1e5
+    model.groupdq[10:, 50:53, 50:53] = pixel.SATURATED
+
+    m_image = RampFitStep.call(
+        model,
+        algorithm='likely',
+        override_gain=m_gain,
+        override_readnoise=m_rnoise,
+        expand_large_events = False,
+    )
+    assert np.std(m_image.data) < 1e-5
+    n_jump_original = np.sum(m_image.dq == pixel.JUMP_DET)
+
+    # Check that the uncertainties are the same for all pixels with jumps
+    # originally flagged.  Check that this uncertainty matches the new value.
+
+    meanerr_jumppixels_orig = np.mean(m_image.err[m_image.dq == pixel.JUMP_DET])
+    assert np.std(m_image.err[m_image.dq == pixel.JUMP_DET]) < 1e-6
+    assert np.abs(meanerr_jumppixels_orig - meanerr_jumppixels_new) < 1e-5
+
+    assert n_jump_original == 112 and n_jump_expanded > 300 and n_jump_expanded < 600
 
 
 def test_uniformweighting():

--- a/romancal/ramp_fitting/tests/test_ramp_fit_likelihood.py
+++ b/romancal/ramp_fitting/tests/test_ramp_fit_likelihood.py
@@ -1,7 +1,5 @@
 """Ramp Fitting tests involving MultiAccum Tables"""
 
-import sys
-
 import numpy as np
 import pytest
 from roman_datamodels.dqflags import pixel
@@ -80,7 +78,7 @@ def test_bad_readpattern():
         )
 
 
-@pytest.mark.skip(reason='See rcal-1324')
+@pytest.mark.skip(reason="See rcal-1324")
 def test_flag_large_events_withsnowball():
     """Test that large events are flagged"""
     resultants = create_linear_ramp(n_resultants=20, nrows=100, ncols=100)

